clouddrift/analysis.py

```diff
@@ -147,8 +150,13 @@ def segment(
     array([2, 2, 2, 2])
     """
 
+    if type(tolerance) == timedelta:
+        positive_tol = tolerance > timedelta(seconds=0)
+    else:
+        positive_tol = tolerance > 0
```

Comment:
Don't you need >= here? Otherwise `tolerance=0` would produce incorrect results, it seems.

```suggestion
    if type(tolerance) == timedelta:
        positive_tol = tolerance >= timedelta(seconds=0)
    else:
        positive_tol = tolerance >= 0
```

Reply:
Let's also add a unit test for tolerance of positive and negative zero.

Reply:
Positive or negative 0?

Reply:
If tolerance is 0, it should just return `np.arange(x)` no? Maybe it should be an ArgumentError.

Reply:
Please ignore my mention of a negative zero. I did some reading and see that both the IEEE standard and Python define signed zero such that `-0 == 0`.

If tolerance is 0, we tolerate consecutive values equal to each other to be a segment. So, `segment([1, 1, 2, 2], tolerance=0)` should return `[2, 2]`.

Reply:
That's a good example right there to form a rowsize variable if it does not exist.